contracts/contracts/mocks/MockHatsProtocol.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import { IHats } from "../interfaces/IHats.sol";
+
+/// @title MockHatsProtocol
+/// @notice A mock contract to test the HatsGatekeeper
+contract MockHatsProtocol {
+  IHats public immutable hats;
+
+  uint256 public lastTopHat;
+  uint256 public lastHat;
+
+  /// @notice Deploy an instance of MockHatsProtocol
+  /// @param _hats The Hats Protocol contract
+  constructor(address _hats) payable {
+    hats = IHats(_hats);
+  }
+
+  /// @notice Creates and mints a Hat that is its own admin, i.e. a "topHat"
+  /// @dev A topHat has no eligibility and no toggle
+  /// @param _target The address to which the newly created topHat is minted
+  /// @param _details A description of the Hat [optional]. Should not be larger than 7000 bytes
+  ///                 (enforced in changeHatDetails)
+  /// @param _imageURI The image uri for this top hat and the fallback for its
+  ///                  downstream hats [optional]. Should not be large than 7000 bytes
+  ///                  (enforced in changeHatImageURI)
+  function mintTopHat(address _target, string calldata _details, string calldata _imageURI) external {
```

Comment:
if this is the same as in `IHats.sol`, then could use `@inheritdoc IHats`, while also add `is IHats` to this contract